### PR TITLE
Fixed LOG_TIMED and ROS_INFO_THROTTLE macros in MODE_GLOG

### LINKED ISSUE #25: Compile error with timed logging functions
- `ROS_INFO_THROTTLE` and `LOG_EVERY_T` throws a compile error in `MODE_LPP` other modes could also be affected.

### PR DESCRIPTION
**Pull request by 4c3y**
- Closes #25 and #24 
- Added Unittests

### CODE REVIEW

File: include/log++.h [hunk 1]
```diff
@@ -617,21 +674,31 @@ class InternalCondLog : public InternalLog {
...
- protected:
-  explicit LogPolicy(int max) : max_(max) {}
+};
+
+template<typename T>
+class LogPolicy : public LogPolicyBase {
+public:
+  explicit LogPolicy(T max) : max_(max) {}
+protected:
+  T max_{0};
```
Comment (michaelpantic) [inline include/log++.h:689]: can T always be initalized with 0? Probably makes sense as its supposed to be float/int, correct? Could add a static assert in the class to make sure it terminates with a good warning if it doesn't work.

File: include/log++.h [hunk 2]
```diff
@@ -617,21 +674,31 @@ class InternalCondLog : public InternalLog {
...
+};
+
+class CountableLogPolicy : public LogPolicy<int> {
+public:
+  explicit CountableLogPolicy(int max) : LogPolicy(max) {}
```
Comment (michaelpantic) [inline include/log++.h:694]: uint?

**Review by michaelpantic — APPROVED** [review]
LGTM, see minor comments. Can be merged once fixed  (if it needs fixing).
Thanks!

File: include/log++.h (hunk 1 above)
Comment (4c3y) [reply include/log++.h:689]: There is a `static_assert` defined for that in Line 815-819. The `InternalLogCount::update()` method calls `InternalLogCount::updatePolicy()`, which is the only method where classes that inherit `LogPolicy<T>` are constructed. Therefore, `T` is guarded by the `static_assert`. `T` always has to be a numeric type and therefore can always be initialized with 0.

**Pull request merged**